contracts/integrations/test/framework/assertions/finalizePool.ts

```diff
@@ -0,0 +1,269 @@
+
+const PRECISION = 15;
+const ALPHA_NUMERATOR = 1;
+const ALPHA_DENOMINATOR = 3;
+const COBB_DOUGLAS_ALPHA = toDecimal(ALPHA_NUMERATOR).dividedBy(toDecimal(ALPHA_DENOMINATOR));
+
+// Reference function for Cobb-Douglas
+function cobbDouglas(poolStats: PoolStats, aggregatedStats: AggregatedStats): BigNumber {
+    const { feesCollected, weightedStake } = poolStats;
+    const { rewardsAvailable, totalFeesCollected, totalWeightedStake } = aggregatedStats;
+
+    const feeRatio = toDecimal(feesCollected).dividedBy(toDecimal(totalFeesCollected));
+    const stakeRatio = toDecimal(weightedStake).dividedBy(toDecimal(totalWeightedStake));
+    // totalRewards * feeRatio ^ alpha * stakeRatio ^ (1-alpha)
+    return new BigNumber(
+        feeRatio
+            .pow(COBB_DOUGLAS_ALPHA)
+            .times(stakeRatio.pow(toDecimal(1).minus(COBB_DOUGLAS_ALPHA)))
+            .times(toDecimal(rewardsAvailable))
+            .toFixed(0, BigNumber.ROUND_FLOOR),
+    );
+}
+
+interface FinalizePoolBeforeInfo {
+    poolStats: PoolStats;
+    aggregatedStats: AggregatedStats;
+    poolRewards: BigNumber;
+    cumulativeRewardsLastStored: BigNumber;
+    mostRecentCumulativeRewards: {
+        numerator: BigNumber;
+        denominator: BigNumber;
+    };
+}
+
+/**
+ * Returns a FunctionAssertion for `finalizePool` which assumes valid input is provided. The `after`
+ * callback below is annotated with the solidity source of `finalizePool`.
+ */
+/* tslint:disable:no-unnecessary-type-assertion */
+export function validFinalizePoolAssertion(
+    deployment: DeploymentManager,
+    simulationEnvironment: SimulationEnvironment,
+): FunctionAssertion<[string], FinalizePoolBeforeInfo, void> {
+    const { stakingWrapper } = deployment.staking;
+
+    return new FunctionAssertion<[string], FinalizePoolBeforeInfo, void>(stakingWrapper, 'finalizePool', {
+        before: async (args: [string]) => {
+            const [poolId] = args;
+            const { currentEpoch } = simulationEnvironment;
+            const prevEpoch = currentEpoch.minus(1);
+
+            const poolStats = PoolStats.fromArray(await stakingWrapper.poolStatsByEpoch(poolId, prevEpoch).callAsync());
+            const aggregatedStats = AggregatedStats.fromArray(
+                await stakingWrapper.aggregatedStatsByEpoch(prevEpoch).callAsync(),
+            );
+            const poolRewards = await stakingWrapper.rewardsByPoolId(poolId).callAsync();
+            const [
+                mostRecentCumulativeRewards,
+                cumulativeRewardsLastStored,
+            ] = await stakingWrapper.getMostRecentCumulativeReward(poolId).callAsync();
+            return {
+                poolStats,
+                aggregatedStats,
+                poolRewards,
+                cumulativeRewardsLastStored,
+                mostRecentCumulativeRewards,
+            };
+        },
+        after: async (beforeInfo: FinalizePoolBeforeInfo, result: FunctionResult, args: [string]) => {
```

Comment:
Optional nit - could be more readable to split this function up. It's a couple hundy lines long.